doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,1151 @@
+        - Set these nexthops as part of the alternative path by updating the SAI attributes for the nexthop group or member accordingly.
+4. RouteOrch will create NHG with ARS object id only when ArsOrch identify the prefix as ARS enabled.
+    * If SAI returns SAI_STATUS_TABLE_FULL or SAI_STATUS_INSUFFICIENT_RESOURCES, RouteOrch will attempt
+      reprogramming of NHG without ARS object, if ARS enabled.
+
+##### __Figure 7: Nexthop group table created by NhgOrch flow__
+![](images/nhg_create_by_nhgo.png)
+
+1. Add NEXTHOP_GROUP_TABLE Table
+    * The APPL_DB adds the NEXTHOP_GROUP_TABLE table, triggering the process.
+2. Identification of ARS-enabled NHG
+    * NhgOrch component will call ArsOrch function when NHG is added/modified.
+    * ARS orch reads the "ars-nhg_path_selector_mode" from ARS_PROFILE and determines ARS enablement for NHG based on configured mode,
+        - In global based mode, system-wide selection.
+        - In interface-based mode, only the interfaces will be matched.
+        - In nexthop-based mode, both the nexthops and the interfaces will be matched.
+3. Nexthop group's SAI attributes handling:
+    * If nexthop group's ARS object ID "set" capability is supported:
+        - The nexthop group is updated with the ARS object ID.
+    * If "set" capability is not supported - ignore NHG update.
+    * If an alternative path is defined:
+        - Identify the relevant nexthops by matching them against the entries in the ARS_NEXTHOPS table with the role set to "alternative_path."
+        - Set these nexthops as part of the alternative path by updating the SAI attributes for the nexthop group or member accordingly.
+4. Add ROUTE_TABLE Table
+    * The APPL_DB adds the ROUTE_TABLE table, triggering the process.
+5. Route creation
+    * RouteOrch creates/updates route with NHG reference.
+
+##### __Figure 8: ARS LAG table creation flow__
+![](images/ars_lag_create.png)
+
+1. Add ARS_OBJECT Table
+    * The CONFIG_DB adds the ARS_OBJECT table, triggering the process.
+2. Identification of ARS-enabled LAG
+    * ARS will determine whether ARS_PROFILE's ars_lag_path_selector_mode is global-based or interface-based.
+        - In interface-based mode, only the interfaces will be matched.
+2. Add ARS_PORTCHANNELS Table 
+    * The CONFIG_DB adds to enables ARS capabilities for the LAG and to associate LAG to alternative_path.
+      ARS_PORTCHANNELS is mandatory when ARS_PROFILE's ars_lag_path_selector_mode is interface-based.
+3. Create SAI ARS Object
+    * The ArsOrch component receives the update and creates the SAI ARS object.
+4. LAG's SAI attributes handling:
+    * ARS checks the state of the LAG members interface's ARS capability and proceeds only if ARS is enabled.
+    * If LAG was created:
+        - It is updated with the ARS object ID.
+
+##### __Figure 9: LAG creation flow__
+![](images/lag_create.png)
+
+1. LAG creation
+    * PortsOrch receives LAG from APPL_DB, triggering the process.
+2. Notification

```

Comment:
Do we still have event notification between orchagent modules? as discussed in the meeting earlier, we can simply make function calls among orchagent modules. please fix it.